### ISSUE TITLE
Ambiguous call in sdl2.nim

### ISSUE THREAD

**Issue by Puasonych** [ftsf/nico#9]
Hello!
I'm trying to compile the following code for Windows.
(Nim Compiler Version 0.18.1 [Windows: amd64])

> nim cpp --nilseqs:on -r main.nim

```
import nico

proc game_init() =
    echo "init"

proc game_update(dt: Pfloat) = 
    # echo "update: ", dt
    discard

proc game_draw() =
    cls()
    setColor(7)
    print("hello world", 42, 60)

nico.init("test", "test_game")

nico.createWindow("test_game", 128, 128, 4)

nico.run(game_init, game_update, game_draw)
```

I get an error:

> C:\Users\Erik\.nimble\pkgs\nico-0.1.0\nico\backends\sdl2.nim(1134, 16) Error: ambiguous call; both system.$(x: int)[declared in lib\system.nim(1864, 5)] and system.$(x: uint64)[declared in lib\system\strmantle.nim(282, 5)] match for: (uint8)

**Comment by uvtechs** [comment]
Hello I had the same issue when trying to compile a game using the nico framework on linux x86. Im new to the nim language and was a little anxious to try it out.

The error is complaining about not being able to differentiate which string conversion function to call? Maybe a custom $ is overloaded or is needed. uint8 seems to be the cause of the confusion and is prob a custom type. Either way I haven't bumped my head on tilemaps yet so until then I punted the details of a proper fix so I could complete my small game using nico.

Anyway I was able to resolve the error message and compile my nico application after patching over the offending line in backends/sdl2.nim.

The offending line on my system is 1134, within the saveMap*(filename: string) procedure.
Change the line from:
fs.write(t)
To:
fs.write(t.char)

What repercussions this may have I can't say as I am completely new to the nim language and nico framework. That being said I have been happily hacking away on my game using the nico framework for the past week. Which is def nice as I was frustrated initially.
